### PR TITLE
KeyFile Migration Fixes

### PR DESCRIPTION
**Pull request by brthor**
Fixes Key Signing migration including the non-windows workaround.

@eerhardt

### CODE REVIEW

File: src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs [hunk 1]
```diff
@@ -26,12 +26,17 @@ private AddPropertyTransform<CommonCompilerOptions>[] EmitEntryPointTransforms
...
+                new AddPropertyTransform<CommonCompilerOptions>("KeyOriginatorFile",
                     compilerOptions => compilerOptions.KeyFile,
                     compilerOptions => !string.IsNullOrEmpty(compilerOptions.KeyFile)),
                 new AddPropertyTransform<CommonCompilerOptions>("SignAssembly",
                     "true",
                     compilerOptions => !string.IsNullOrEmpty(compilerOptions.KeyFile))
+                    .WithMSBuildCondition(" '$(OS)' == 'Windows_NT' "),
+                new AddPropertyTransform<CommonCompilerOptions>("PublicSign", 
+                    "true",
+                    compilerOptions => !string.IsNullOrEmpty(compilerOptions.KeyFile))
```
Comment (eerhardt) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs]: Should also include a condition if publicSign isn't explicitly defined in the project.json.

File: TestAssets/TestProjects/TestAppWithSigning/project.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "keyFile": "./key.snk"
+  },
+  "dependencies": {
+    "Microsoft.NETCore.App": "1.0.1"
+  },
+  "frameworks": {
+    "netcoreapp1.0": {}
+  },
+  "runtimes": {
+    "win7-x64": {},
```
Comment (eerhardt) [inline TestAssets/TestProjects/TestAppWithSigning/project.json]: Any reason why this needs to be a self-contained app?  Portable would be more simple, and wouldn't need to restore the runtime assets.

File: test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs [hunk 3]
```diff
@@ -26,6 +26,7 @@ public class GivenThatIWantToMigrateTestApps : TestBase
         // https://github.com/dotnet/sdk/issues/73 [InlineData("TestAppWithLibrary/TestApp", false)]
         [InlineData("TestAppWithRuntimeOptions")]
         [InlineData("TestAppWithContents")]
+        [InlineData("TestAppWithSigning")]
```
Comment (eerhardt) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs]: Do we do any verification in this test that the assembly is signed?

File: TestAssets/TestProjects/TestAppWithSigning/project.json (hunk 2 above)
Comment (brthor) [inline TestAssets/TestProjects/TestAppWithSigning/project.json]: copy pasta, fixed

File: test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs (hunk 3 above)
Comment (brthor) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs]: We do a test that the properties that get passed csc for signing are populated. 

How do we test that the assembly is signed?

Comment (brthor) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs]: fixed.

**Comment by eerhardt** [comment]
:shipit:

File: src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs (hunk 1 above)
Comment (brthor) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs]: fixed.

**Pull request merged**